ports/servoshell/app.rs

```diff
@@ -40,8 +40,13 @@ pub struct App {
 
 /// Action to be taken by the caller of [`App::handle_events`].
 enum PumpResult {
+    /// The caller should shutdown Servo and its related contexnt.
     Shutdown,
+    /// A new frame is ready to present. The caller can paint other things themselves during this
+    /// period, but has to call [`Servo::present`] to tell Servo compositor continue rendering.
     ReadyToPresent,
+    /// The size has changed. The caller can paint other things themselves during this
+    /// period, but has to call [`Servo::present`] to tell Servo compositor continue rendering.
```

Comment:
suggest “to perform the page flip and tell the Servo compositor to continue rendering”, because these are two separate things that are currently coupled but we may change later